pkg/operator/controller/gatewayclass/controller.go

```diff
@@ -90,15 +123,29 @@ func (r *reconciler) Reconcile(ctx context.Context, request reconcile.Request) (
 
 	var gatewayclass gatewayapiv1.GatewayClass
 	if err := r.cache.Get(ctx, request.NamespacedName, &gatewayclass); err != nil {
+		log.Error(err, "failed to get gatewayclass", "request", request)
 		return reconcile.Result{}, err
```

Comment:
Errors returned from `Reconcile` are logged as errors by the controller-runtime. Failure to get a resource will be clear enough straight out of the client.
```suggestion
		return reconcile.Result{}, err
```